<launch>
    <!-- Provide the robot_description in the standard urdf format, but with
    the shared link split up to produce a number of chains. -->
    <param name="robot_description"
        command="xacro '$(find cga_kinematics)/model/delta.urdf.xacro'"/>

    <rosparam file="$(find cga_kinematics)/config/loops.yaml" />

    <!-- Start joint_state_publisher_gui to create a gui for controlling
    joint angles -->
    <node
        pkg="joint_state_publisher_gui"
        type="joint_state_publisher_gui"
        name="joint_state_publisher_gui">
        <remap from="joint_states" to="joint_states_in"/>
    </node>

    <node
        pkg="cga_kinematics"
        type="delta_fk"
        name="delta_fk"
        output="screen">
    </node>

    <node
        pkg="robot_state_publisher"
        type="robot_state_publisher"
        name="robot_state_publisher">
        <remap from="joint_states" to="joint_states_out"/>
    </node>

    <node
        name="rviz" pkg="rviz" type="rviz"
        args="-d $(find cga_kinematics)/config/default.rviz"
    />
</launch>
